src/overlays/effects/ovl_Effect_Ss_En_Fire/z_eff_ss_en_fire.c

```diff
@@ -5,23 +5,144 @@
  */
 
 #include "z_eff_ss_en_fire.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+
+#define rScaleMax regs[0]
+#define rScale regs[1]
+#define rLifespan regs[2]
+#define rUnused regs[3]
+#define rPitch regs[4]
+#define rYaw regs[5]
+#define rReg6 regs[6]
+#define rBodyPart regs[7]
+#define rFlags regs[8]
+#define rScroll regs[9]
 
 #define PARAMS ((EffectSsEnFireInitParams*)initParamsx)
 
-s32 EffectSsEnFire_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
+u32 EffectSsEnFire_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
 void EffectSsEnFire_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsEnFire_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
 const EffectSsInit Effect_Ss_En_Fire_InitVars = {
     EFFECT_SS_EN_FIRE,
     EffectSsEnFire_Init,
 };
 
-#endif
+u32 EffectSsEnFire_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsEnFireInitParams* initParams = PARAMS;
+
+    Math_Vec3f_Copy(&this->pos, &initParams->pos);
+    Math_Vec3f_Copy(&this->velocity, &gZeroVec3f);
+    Math_Vec3f_Copy(&this->accel, &gZeroVec3f);
+    this->life = 20;
+    this->rLifespan = this->life;
+    this->actor = initParams->actor;
+    this->rScroll = Rand_ZeroOne() * 20.0f;
+    this->draw = EffectSsEnFire_Draw;
+    this->update = EffectSsEnFire_Update;
+    this->rUnused = -15;
+
+    if (initParams->bodyPart < 0) {
+        this->rYaw = Math_Vec3f_Yaw(&initParams->actor->world.pos, &initParams->pos) - initParams->actor->shape.rot.y;
+        this->rPitch =
+            Math_Vec3f_Pitch(&initParams->actor->world.pos, &initParams->pos) - initParams->actor->shape.rot.x;
+        this->vec.z = Math_Vec3f_DistXYZ(&initParams->pos, &initParams->actor->world.pos);
+    }
+
+    this->rScaleMax = initParams->scale;
+
+    if (initParams->params & ENFIRE_PARAMS_USE_SCALE) {
+        this->rScale = initParams->scale;
+    } else {
+        this->rScale = 0;
+    }
+
+    this->rReg6 = initParams->params & 0x7FFF;
+    this->rBodyPart = initParams->bodyPart;
+    this->rFlags = initParams->flags;
+
+    return 1;
+}
+
+void EffectSsEnFire_Draw(PlayState* play, u32 index, EffectSs* this) {
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+    f32 scale;
+    s16 camYaw;
+    s32 pad[3];
```

Comment:
The number of pads in these functions is interesting, have you noticed if the stack variables might be common between effects?

Reply:
Nothing that really comes to mind. One should be the play cast, but for the other 2 🤷